VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/Strings/StringPlaceholdersInWrongOrder/StringPlaceholdersInWrongOrderAnalyzer.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Collections.Immutable;
+using System.Globalization;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace VSDiagnostics.Diagnostics.Strings.StringPlaceholdersInWrongOrder
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class StringPlaceholdersInWrongOrderAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly string Category = VSDiagnosticsResources.StringsCategory;
+        private const string DiagnosticId = nameof(StringPlaceholdersInWrongOrderAnalyzer);
+        private static readonly string Message = VSDiagnosticsResources.StringPlaceholdersInWrongOrderMessage;
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Warning;
+        private static readonly string Title = VSDiagnosticsResources.StringPlaceholdersInWrongOrderTitle;
+
+        internal static DiagnosticDescriptor Rule => new DiagnosticDescriptor(DiagnosticId, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.InvocationExpression);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = context.Node as InvocationExpressionSyntax;
+            if (invocation == null)
+            {
+                return;
+            }
+
+            // Verify we're dealing with a string.Format() call
```

Comment:
Instead of hard-coding for `string.Format`, look for any invocation of a method which takes a `string format` followed by either a `params object[]` or a series of `object`s.

This will also catch methods like `Console.WriteLine`, `StreamWrite.Write`, `Debug.WriteLine`, etc.